### PR TITLE
INTMDB-609: Add secret manager profile support to Org-invitation resource

### PR DESCRIPTION
**Pull request by maastha**
## Description

Updated Org-invitation CFN to read API keys from secret manager instead of passing API keys.

Link to any related issue(s): 

## Type of change:

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run `make fmt` and formatted my code

## Further comments
```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================= 8 passed, 4 skipped, 12 deselected, 1 warning in 72.70s (0:01:12) =========================================================
```

### CODE REVIEW

File: examples/org-invitation/org-invitation.json [hunk 1]
```diff
@@ -2,16 +2,20 @@
   "AWSTemplateFormatVersion" : "2010-09-09",
   "Description" : "This template creates a invitation at Atlas Organization level",
   "Parameters" : {
-    "Username" : {
-      "Type" : "String",
-      "Description" : "Email address of the MongoDB Cloud user invited to join the organization."
+    "Profile": {
+      "Type": "String",
```
Comment (colm-quinn) [inline examples/org-invitation/org-invitation.json:6]:
```suggestion
      "Description": "Your MongoDB Atlas Profile Name created in secret manager",
      "Type": "String",
```

**Comment by colm-quinn** [comment]
I really like what @andreaangiolillo did for manual validation of the changes before he merged: https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/328#issuecomment-1439856034

Let's do that for these new features going forward, until we have automated QA in place.

**Comment by maastha** [comment]
> I really like what @andreaangiolillo did for manual validation of the changes before he merged: [#328 (comment)](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/328#issuecomment-1439856034)
> 
> Let's do that for these new features going forward, until we have automated QA in place.

Yes currently doing that before merging for all resources, will add more info once complete.

**Comment by maastha** [comment]
Manually tested in AWS via private registry in prod:
Screenshot 2023-02-22 at 14 20 38
Screenshot 2023-02-22 at 14 20 29

File: examples/org-invitation/org-invitation.json (hunk 1 above)
Comment (maastha) [reply examples/org-invitation/org-invitation.json:6]: updated

**Review by colm-quinn — APPROVED** [review]
LGTM!

File: cfn-resources/org-invitation/mongodb-atlas-orginvitation.json [hunk 2]
```diff
@@ -120,18 +117,23 @@
...
-    "/properties/Id"
+    "/properties/Id",
+    "/properties/Profile"
   ],
   "readOnlyProperties": [
     "/properties/ExpiresAt",
-    "/properties/Id", 
+    "/properties/Id",
     "/properties/CreatedAt",
     "/properties/InviterUsername"
```
Comment (andreaangiolillo) [inline cfn-resources/org-invitation/mongodb-atlas-orginvitation.json:131]: have you tried to use GetAttr to retrieve these info in a stack? The example that you updated does not use `GetAttr`

Comment (andreaangiolillo) [reply cfn-resources/org-invitation/mongodb-atlas-orginvitation.json:131]: see this https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/340#discussion_r1114456753

**Comment by maastha** [comment]
@andreaangiolillo Fixed and included GetAtt in example and tested in private registry, let me know if this is sufficient:
Screenshot 2023-02-22 at 16 17 48

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: cfn-resources/org-invitation/mongodb-atlas-orginvitation.json (hunk 2 above)
Comment (maastha) [reply cfn-resources/org-invitation/mongodb-atlas-orginvitation.json:132]: Fixed below

**Pull request merged**